helm/hpcc/templates/_helpers.tpl

```diff
@@ -1119,16 +1136,46 @@ Generate instance queue names
   {{- end }}
  {{- end }}
 {{ end -}}
+{{- $stdThorQueues := dict -}}
+{{- $stdThorQueuesWithAux := dict -}}
 {{- range $.Values.thor -}}
  {{- if not .disabled -}}
-- name: {{ .name }}
-  type: thor
-  {{- if hasKey . "prefix" }}
-  prefix: {{ .prefix }}
-  {{- end }}
-  width: {{ mul (.numWorkers | default 1) ( .channelsPerWorker | default 1) }}
- {{- end }}
-{{ end -}}
+  {{- $queueItem := dict "name" .name "type" "thor" -}}
+  {{- if hasKey . "prefix" -}}
+   {{- $_ := set $queueItem "prefix" .prefix -}}
+  {{- end -}}
+  {{- $_ := set $queueItem "width" (mul (.numWorkers | default 1) ( .channelsPerWorker | default 1)) -}}
+  {{- if hasKey . "auxQueues" -}}
+   {{- $_ := set $stdThorQueuesWithAux .name .auxQueues -}}
+  {{- end -}}
+  {{- $_ := set $stdThorQueues .name $queueItem -}}
+ {{- end -}}
+{{- end -}}
+{{- $auxThorQueues := dict -}}
+{{- range $thorNameWithAuxQueues, $auxQueues := $stdThorQueuesWithAux -}}
+ {{ $queueItem := get $stdThorQueues $thorNameWithAuxQueues -}}
+ {{- range $auxQueueName := $auxQueues -}}
+  {{- if (hasKey $stdThorQueues $auxQueueName) -}}
+   {{- $_ := fail (printf "Thor '%s' defines aux queue '%s' that clashes with existing Thor name" $queueItem.name $auxQueueName) -}}
```

Comment:
Please can you check in test cases in `testing\helm\errtests` and a good example in `testing\helm\tests`